src/adapter/graphql/src/queries/auth.rs

```diff
@@ -33,6 +36,55 @@ impl Auth {
 
         Ok(providers)
     }
+
+    #[allow(clippy::unused_async)]
+    #[tracing::instrument(level = "info", name = Auth_relations, skip_all)]
+    #[graphql(guard = "AdminGuard::new()")]
+    async fn relations(
+        &self,
+        ctx: &Context<'_>,
+        account_ids: Vec<AccountID<'_>>,
+    ) -> Result<Vec<AuthRelation>> {
+        let rebac_service = from_catalog_n!(ctx, dyn RebacService);
+
+        let account_ids = account_ids.into_iter().map(Into::into).collect::<Vec<_>>();
+        let authorized_datasets_map = rebac_service
+            .get_authorized_datasets_by_account_ids(&account_ids)
+            .await
+            .int_err()?;
+
+        let mut auth_relations = Vec::new();
+
+        for (account_id, authorized_datasets) in authorized_datasets_map {
+            auth_relations.reserve_exact(authorized_datasets.len());
+
+            for authorized_dataset in authorized_datasets {
+                let gql_account = Account::from_account_id(ctx, account_id.clone()).await?;
```

Comment:
Account is loaded in the inner loop, no reason for it

Reply:
Batch requests added

Fixed